components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2TokenValidationServiceTest.java

```diff
@@ -0,0 +1,198 @@
+
+    @Mock
+    private OAuth2TokenValidationRequestDTO mockedTokenValidationRequestDTO;
+
+    @Mock
+    private TokenValidationHandler mockedValidationHandler;
+
+    @Mock
+    private OAuthComponentServiceHolder mockedOAuthComponentServiceHolder;
+
+    @Mock
+    private OAuthEventInterceptor mockedOAuthEventInterceptor;
+
+    @Mock
+    private OAuth2TokenValidationResponseDTO mockedOAuth2TokenValidationResponseDTO;
+
+    @Mock
+    private OAuth2IntrospectionResponseDTO mockedIntrospectionResponseDTO;
+
+    @Mock
+    private OAuth2TokenValidationRequestDTO mockedOAuth2TokenValidationRequestDTO;
+
+    @Mock
+    private OAuth2ClientApplicationDTO mockedClientApplicationDTO;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        tokenValidationService = new OAuth2TokenValidationService();
+
+        mockStatic(TokenValidationHandler.class);
+        when(TokenValidationHandler.getInstance()).thenReturn(mockedValidationHandler);
+
+        mockStatic(OAuthComponentServiceHolder.class);
+        when(OAuthComponentServiceHolder.getInstance()).thenReturn(mockedOAuthComponentServiceHolder);
+        when(mockedOAuthComponentServiceHolder.getOAuthEventInterceptorProxy()).thenReturn(mockedOAuthEventInterceptor);
+    }
+
+    @Test
+    public void testValidate() throws Exception {
+        when(mockedOAuthEventInterceptor.isEnabled()).thenReturn(true, false);
+
+        when(mockedValidationHandler.validate(any(OAuth2TokenValidationRequestDTO.class))).thenReturn
+                (mockedOAuth2TokenValidationResponseDTO);
+
+        assertNotNull(tokenValidationService.validate(mockedTokenValidationRequestDTO));
+    }
+
+    @Test
+    public void testValidateWithErrorResponse() throws Exception {
+        when(mockedOAuthEventInterceptor.isEnabled()).thenReturn(true);
+        doThrow(new IdentityOAuth2Exception("dummyException")).when(mockedOAuthEventInterceptor).onPreTokenValidation
+                (any(OAuth2TokenValidationRequestDTO.class), anyMap());
+        OAuth2TokenValidationResponseDTO oAuth2TokenValidationResponseDTO = tokenValidationService.validate(mockedTokenValidationRequestDTO);
+
+        assertNotNull(oAuth2TokenValidationResponseDTO);
+        assertEquals(oAuth2TokenValidationResponseDTO.getErrorMsg(), "dummyException");
+    }
+
+    @Test
+    public void testValidateWithErrorResponse2() throws Exception {
+        when(mockedOAuthEventInterceptor.isEnabled()).thenReturn(true);
+
+        doThrow(new IdentityOAuth2Exception("dummyException")).when(mockedValidationHandler).validate(any
+                (OAuth2TokenValidationRequestDTO.class));
+        OAuth2TokenValidationResponseDTO oAuth2TokenValidationResponseDTO = tokenValidationService.validate
+                (mockedTokenValidationRequestDTO);
+
+        assertNotNull(oAuth2TokenValidationResponseDTO);
+        assertEquals(oAuth2TokenValidationResponseDTO.getErrorMsg(), "Server error occurred while validating the OAuth2 access token");
```

Comment:
break into two lines if it exceeds 120 characters

Reply:
done